### PR TITLE
[docs] Fix incorrect gradient support documentation for torch.sparse.mm and sparse.addmm

### PR DESCRIPTION
**Pull request by subinz1**
## Summary

Fixes incorrect documentation for `torch.sparse.mm` and `sparse.addmm` that stated CSR gradients are not supported, when they actually are.

## Changes

The documentation previously stated:
> "This function doesn't support computing derivatives with respect to CSR matrices."

This is **incorrect**. CSR @ Dense operations do support backward pass and return sparse CSR gradients on both CPU and CUDA.

Updated the `.. note::` sections to accurately describe gradient support:

| Operation | Forward | Backward | Gradient Type |
|-----------|---------|----------|---------------|
| COO @ Dense | ✅ | ✅ | sparse COO |
| CSR @ Dense | ✅ | ✅ | sparse CSR |
| COO @ COO | ✅ | ❌ | - |
| CSR @ CSR | ✅ | ❌ | - |
| COO @ CSR | ❌ | - | - |
| CSR @ COO | ❌ | - | - |

## Reproduction

```python
import torch

# CSR @ Dense - works and returns sparse CSR gradient
A = torch.tensor([[1., 0, 2], [0, 3, 0]]).to_sparse_csr().requires_grad_(True)
B = torch.tensor([[0, 1.], [2, 0], [0, 0]], requires_grad=True)
y = torch.sparse.mm(A, B)
y.sum().backward()
print(f"A.grad layout: {A.grad.layout}")  # torch.sparse_csr
```

Fixes #172550

cc @nikitaved @pearu @cpuhrsch @amjames @bhosmer @jcaip @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

File: torch/sparse/__init__.py [hunk 1]
```diff
@@ -47,14 +47,20 @@
 sparse.addmm(mat, mat1, mat2, *, beta=1., alpha=1.) -> Tensor
 
 This function does exact same thing as :func:`torch.addmm` in the forward,
-except that it supports backward for sparse COO matrix :attr:`mat1`.
+except that it supports backward for sparse matrix :attr:`mat1`.
```
Comment (pearu) [inline torch/sparse/__init__.py]:
```suggestion
except that it supports backward for sparse COO and CSR matrix :attr:`mat1`.
```

**Review by pearu — CHANGES_REQUESTED** [review]
Thanks for the PR, @subinz1! This looks good to me.

While autograd of `sparse.{mm,addmm}` supports CSR-dense combination, the support for other sparse compressed tensors such as CSC, BSR, BSC is still missing (see `test_gradcheck_mm` in test/test_sparse.py). Could you include the support state of CSC/BSR/BSC tensors to `addmm` and `mm` documentations as well?

**Review by pearu — APPROVED** [review]
LGTM! Thanks, @subinz1!

**Review by nikitaved — APPROVED** [review]
